baseline/baseline.py

```diff
@@ -0,0 +1,195 @@
+            [-1, 1]. We are making the simplifying assumption that
+            for every task, all sounds will be padded/trimmed to
+            the same length. This doesn’t preclude people from
+            using the API for corpora of variable-length sounds;
+            merely we don’t implement that as a core feature. It
+            could be a wrapper function added later.
+            model: Loaded model, in PyTorch or Tensorflow 2.x. This
+        should be moved to the device the audio tensor is on.
+        hop_size: Number of audio samples between adjacent frames
+            batch_size: The participants are responsible for estimating
+        the batch_size that will achieve high-throughput while
+        maintaining appropriate memory constraints. However,
+        batch_size is a useful feature for end-users to be able to
+        toggle.
+            center: If True, the timestamps correspond to the center
+            of each analysis window. center=True will be used for
+            all evaluation tasks.
+    Returns:
+            ({embedding_size: Tensor}, list(frame timestamps)) where
+            embedding_size can be any of [4096, 2048, 512, 128,
+            20].  Tensor is float32 (or signed int for 20-dim),
+            n_sounds x n_frames x dim.
+
+    """
+    assert audio.ndim == 2
+    model = model.to(audio.device)
+
+    # Implement batching of the audio
+    if batch_size is None:
+        # Here we just pick a sensible default batch size
+        batch_size = 512
+    dataset = torch.utils.data.TensorDataset(audio)
+    loader = torch.utils.data.DataLoader(
+        dataset, batch_size=batch_size, shuffle=False, drop_last=False
+    )
+
+    if center:
+        timestamps = (
+            torch.range(0, audio.shape[1], hop_size_samples) / input_sample_rate()
+        )
+    else:
+        # We will only use center=True in HEAR 2021
+        raise ValueError("center = False not supported")
+
+    # Put the model into eval mode, and don't compute any gradients.
+    model.eval()
+    with torch.no_grad():
+        # Iterate over all batches and accumulate the embeddings
+        # into allembs
+        allembs = defaultdict(list)
+        for batch in loader:
+            # The dataset only has one element, which is the audio
+            # batch tensor
+            embs = model(batch[0], hop_size_samples=hop_size_samples, center=center)
+            for e in embs:
+                allembs[e].append(embs[e])
+    # Concatenate the minibatches before returning
+    # TODO: Check that returns are the right type?
+    for e in allembs:
+        allembs[e] = torch.cat(allembs[e], dim=0)
+        assert allembs[e].shape[0] == audio.shape[0]
+        assert len(timestamps) == allembs[e].shape[1]
+    return allembs, timestamps
+
+
+def pairwise_distance(emb1: Tensor, emb2: Tensor) -> Tensor:
+    """
+    Note that if you are calling this with the 20-dim int8,
+    you should cast them to .float() first.
```

Comment:
Given that we're stipulating the cast to int8, maybe we should take care of this possibility ourselves.